src/transaction_builder.js

```diff
@@ -153,33 +172,252 @@ export class TransactionBuilder {
+    }
+
+    this.minAccountSequence = minAccountSequence;
+
+    return this;
+  }
+
+  /**
+   * For the transaction to be valid, the current ledger time must
+   * be at least minAccountSequenceAge greater than sourceAccount's sequenceTime.
+   * Internally this will set the <code>minAccountSequenceAge</code> precondition.
+   *
+   * @param {number} durationInSeconds The minimum amount of time between source account sequence time and the ledger time when this transaction will become valid.
+   * If the value is `0`, the transaction is unrestricted by the account sequence age. Cannot be negative.
+   * @return {TransactionBuilder}
+   * @see TransactionPreconditions
+   */
+  setMinAccountSequenceAge(durationInSeconds) {
+    if (this.minAccountSequenceAge !== null && this.minAccountSequenceAge > 0) {
+      throw new Error(
+        'min_account_sequence_age has been already set - setting min_account_sequence_age would overwrite it.'
+      );
+    }
+
+    if (durationInSeconds < 0) {
+      throw new Error('min_account_sequence_age cannot be negative');
+    }
+
+    this.minAccountSequenceAge = durationInSeconds;
+
+    return this;
+  }
+
+  /**
+   * For the transaction to be valid, the current ledger number
+    must be at least minAccountSequenceLedgerGap greater than sourceAccount's ledger sequence.
+   * Internally this will set the <code>minAccountSequenceLedgerGap</code> precondition.
+   *
+   * @param {number} gap The minimum number of ledgers between source account sequence and the ledger number when this transaction will become valid.
+   * If the value is `0`, the transaction is unrestricted by the account sequence ledger. Cannot be negative.
+   * @return {TransactionBuilder}
+   * @see TransactionPreconditions
+   */
+  setMinAccountSequenceLedgerGap(gap) {
+    if (
+      this.minAccountSequenceLedgerGap !== null &&
+      this.minAccountSequenceLedgerGap > 0
+    ) {
+      throw new Error(
+        'min_account_sequence_ledger_gap has been already set - setting min_account_sequence_ledger_gap would overwrite it.'
+      );
+    }
+
+    if (gap < 0) {
+      throw new Error('min_account_sequence_ledger_gap cannot be negative');
+    }
+
+    this.minAccountSequenceLedgerGap = gap;
+
+    return this;
+  }
+
+  /**
+   * For the transaction to be valid, there must be a signature
+   * corresponding to every Signer in this array, even if the
+   * signature is not otherwise required by the sourceAccount or
+   * operations.
+   * Internally this will set the <code>extraSigners</code> precondition.
+   *
+   * @param {string[]} extraSigners The required extra signers.
+   * @return {TransactionBuilder}
+   * @see TransactionPreconditions
+   */
+  setExtraSigners(extraSigners) {
+    if (this.extraSigners !== null && this.extraSigners.length > 0) {
+      throw new Error(
+        'extra_signers has been already set - setting extra_signers would overwrite it.'
+      );
+    }
+
+    this.extraSigners = extraSigners;
```

Comment:
Do we copy the array? Dunno

Reply:
Also can sanity check that `extraSigners.length <= 2` since more will blow up in the XDR encode step